chia/rpc/wallet_rpc_api.py

```diff
@@ -1445,25 +1446,26 @@ async def dl_update_multiple(self, request) -> Dict:
 
         for _, wallet in self.service.wallet_state_manager.wallets.items():
             if WalletType(wallet.type()) == WalletType.DATA_LAYER:
-                # TODO: This method should optionally link the singletons with announcements.
-                #       Otherwise spends are vulnerable to signature subtraction.
-                tx_records: List[TransactionRecord] = []
-                for launcher, root in request["updates"].items():
-                    records = await wallet.create_update_state_spend(
-                        bytes32.from_hexstr(launcher), bytes32.from_hexstr(root)
-                    )
-                    tx_records.extend(records)
-                # Now that we have all the txs, we need to aggregate them all into just one spend
-                modified_txs: List[TransactionRecord] = []
-                aggregate_spend = SpendBundle([], G2Element())
-                for tx in tx_records:
-                    if tx.spend_bundle is not None:
-                        aggregate_spend = SpendBundle.aggregate([aggregate_spend, tx.spend_bundle])
-                        modified_txs.append(dataclasses.replace(tx, spend_bundle=None))
-                modified_txs[0] = dataclasses.replace(modified_txs[0], spend_bundle=aggregate_spend)
-                for tx in modified_txs:
-                    await self.service.wallet_state_manager.add_pending_transaction(tx)
-                return {"tx_records": [rec.to_json_dict_convenience(self.service.config) for rec in modified_txs]}
+                async with self.service.wallet_state_manager.lock:
+                    # TODO: This method should optionally link the singletons with announcements.
+                    #       Otherwise spends are vulnerable to signature subtraction.
+                    tx_records: List[TransactionRecord] = []
+                    for launcher, root in request["updates"].items():
+                        records = await wallet.create_update_state_spend(
+                            bytes32.from_hexstr(launcher), bytes32.from_hexstr(root)
+                        )
+                        tx_records.extend(records)
```

Comment:
Just because I usually see it avoided here, this silently assumes that `records` has to be non-None yeah?

Reply:
It's not so much an assumption as a fact of the code: https://github.com/Chia-Network/chia-blockchain/blob/c231ad449b9be6cf08a6e2548bbe262e08aeb9b1/chia/data_layer/data_layer_wallet.py#L428

_Trust me_ mypy would be letting me know if it could possibly be `None`